sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1341,6 +1341,16 @@ class SessionCatalog(
     functionRegistry.registerFunction(func, info, builder)
   }
 
+  /**
+   * Unregister a temporary or permanent function from a session-specific [[FunctionRegistry]]
+   */
+  def unregisterFunction(name: FunctionIdentifier): Unit = {
+    if (!functionRegistry.dropFunction(name)) {
+      throw new NoSuchFunctionException(
```

Comment:
Actually, it does not throw this exception because this check's already done in https://github.com/apache/spark/pull/28840/files#diff-d2a203f08c862bd762e6740c16e972f7R267-R268 ?